src/hooks/useLocalStorage.ts

```diff
@@ -0,0 +1,23 @@
+import { useState, useEffect } from 'react'
+
+function getStorageValue(key: string, defaultValue: string) {
+    // getting stored value
+    if (typeof window !== 'undefined') {
+        const saved = localStorage.getItem(key)
+        const initial = saved !== null ? JSON.parse(saved) : defaultValue
+        return initial
+    }
+}
+
+export const useLocalStorage = (key: string, defaultValue = '') => {
+    const [value, setValue] = useState(() => {
+        return getStorageValue(key, defaultValue)
+    })
+
+    useEffect(() => {
+        // storing input name
+        localStorage.setItem(key, JSON.stringify(value))
```

Comment:
This will always perform the item setting. Not sure if that's desired. If you want to avoid that you can use a little trick – `useRef`.

```ts
const isUpdateRef = useRef<boolean>(false)

useEffect(() => {
    if (!isUpdateRef.current) {
        isUpdateRef.current = true
        // It's a on-mount run. Ignore the rest.
        return
    }

    // The rest. Here, things that happen on an actual update.
}, [value/*, setValue (redundant!)*/])
```